recipes/wolfssl/all/conanfile.py

```diff
@@ -63,9 +64,8 @@ def configure(self):
         del self.settings.compiler.libcxx
 
     def build_requirements(self):
-        if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH") and \
-                tools.os_info.detect_windows_subsystem() != "msys2":
-            self.build_requires("msys2/20190524")
+        if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH"):
```

Comment:
Imho, the check `tools.os_info.detect_windows_subsystem() != "msys2"` should be removed from all recipes.
When using Github Actions on Windows, the bash shell included with Git for Windows is available.

This bash shell is severely limited and causes building wolfssl to fail with the following error:
```
---
wolfssl/4.5.0: run_in_windows_bash: "C:\Program Files\Git\bin\bash.exe" --login -c ^"cd \^"/c/users/runneradmin/.conan/data/wolfssl/4.5.0/_/_/build/1eae37476b4eeff993cab96534a6ad09f8fd04d4\^" ^&^& PATH=\^"/c/users/runneradmin/.conan/data/libtool/2.4.6/_/_/package/82ad5af1557e85f11dfc0f68ec2820c275dfe6d4/bin:/c/users/runneradmin/.conan/data/automake/1.16.2/_/_/package/3e48e69237f7f2196164383ef9dedf0f93cbf249/bin:/c/users/runneradmin/.conan/data/autoconf/2.69/_/_/package/ca33edce272a279b24f87dc0d4cf5bbdcffbc187/bin:/c/users/runneradmin/.conan/data/m4/1.4.18/_/_/package/3fb49604f9c2f729b85ba3115852006824e72cab/bin:$PATH\^" ^&^& make -j2 ^"

----Running------
> "C:\Program Files\Git\bin\bash.exe" --login -c ^"cd \^"/c/users/runneradmin/.conan/data/wolfssl/4.5.0/_/_/build/1eae37476b4eeff993cab96534a6ad09f8fd04d4\^" ^&^& PATH=\^"/c/users/runneradmin/.conan/data/libtool/2.4.6/_/_/package/82ad5af1557e85f11dfc0f68ec2820c275dfe6d4/bin:/c/users/runneradmin/.conan/data/automake/1.16.2/_/_/package/3e48e69237f7f2196164383ef9dedf0f93cbf249/bin:/c/users/runneradmin/.conan/data/autoconf/2.69/_/_/package/ca33edce272a279b24f87dc0d4cf5bbdcffbc187/bin:/c/users/runneradmin/.conan/data/m4/1.4.18/_/_/package/3fb49604f9c2f729b85ba3115852006824e72cab/bin:$PATH\^" ^&^& make -j2 ^"
-----------------
mingw32-make: *** No rule to make target '/c/users/runneradmin/.conan/data/wolfssl/4.5.0/_/_/build/1eae37476b4eeff993cab96534a6ad09f8fd04d4/source_subfolder/Makefile.am', needed by '/c/users/runneradmin/.conan/data/wolfssl/4.5.0/_/_/build/1eae37476b4eeff993cab96534a6ad09f8fd04d4/source_subfolder/Makefile.in'.  Stop.

wolfssl/4.5.0: ERROR: Package '1eae37476b4eeff993cab96534a6ad09f8fd04d4' build failed
wolfssl/4.5.0: WARN: Build folder C:\Users\runneradmin\.conan\data\wolfssl\4.5.0\_\_\build\1eae37476b4eeff993cab96534a6ad09f8fd04d4
wolfssl/4.5.0: 
ERROR: wolfssl/4.5.0: Error in build() method, line 132
	autotools.make()
	ConanException: Error 2 while executing make -j2
Error: Process completed with exit code 1.
```

Reply:
It would even better if the entire logic with requiring msys2 is moved into Conan itself like `AutoToolsBuildEnvironment` or `tools.run_in_windows_bash`. Too much boilerplate and hardcoded versions in recipes...

Reply:
agree that such repetitive patterns should be built-in. especially error prone, like msys2 detection.
this is subject to discuss, so feel free to open conan client issue.
e.g. some users already have full-feature msys2 installation and don't want to waste time, space and traffic to download one more msys archive from conan.
another users have only limited bash from Git installed and just want things to work out of the box.
maybe detection should be able to distinguish bash from Git and MSYS2? 
like if Git bash is somehow incomplete, what exactly is missing to make auto-tools work in it? use this tool/feature in conan detection then.